xarray/backends/api.py

```diff
@@ -876,16 +876,23 @@ def to_zarr(dataset, store=None, mode='w-', synchronizer=None, group=None,
     _validate_dataset_names(dataset)
     _validate_attrs(dataset)
 
-    store = backends.ZarrStore.open_group(store=store, mode=mode,
+    zstore = backends.ZarrStore.open_group(store=store, mode=mode,
                                           synchronizer=synchronizer,
                                           group=group)
 
     writer = ArrayWriter()
     # TODO: figure out how to properly handle unlimited_dims
-    dump_to_store(dataset, store, writer, encoding=encoding)
+    dump_to_store(dataset, zstore, writer, encoding=encoding)
     writes = writer.sync(compute=compute)
 
     if not compute:
         import dask
-        return dask.delayed(_finalize_store)(writes, store)
-    return store
+        return dask.delayed(_finalize_store)(writes, zstore)
+        # TODO: handle consolidation for delayed case
+
+    if consolidate:
+        import zarr
+        zarr.consolidate_metadata(store)
+        # do we need to reload ztore now that we have consolidated?
```

Comment:
would it make sense for zarr to handle this?

Reply:
What do you mean?

Reply:
I meant reloading the zarr store automatically.

Reply:
I think that would be hard to achieve. And I'm not sure it's necessary. Frankly I don't know why we return a store object from `to_zarr` at all.

Reply:
`zarr.consolidate_metadata` returns the output of open_consolidated on the same store, so this is already happening